### PR TITLE
Add metrics for min/max priority fee per slot, and counters for fee/non-fee transactions

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem
Would like to add min/max fee to [fee_market](https://metrics.solana.com:3000/d/0n54roOVz/fee-market?orgId=1) dashboard, as well as counts of transactions with and without fee.

#### Summary of Changes
- add counters

Fixes #

### CODE REVIEW

File: runtime/src/prioritization_fee.rs [hunk 1]
```diff
@@ -125,6 +173,7 @@ impl PrioritizationFee {
 
                 self.metrics
                     .accumulate_total_prioritization_fee(transaction_fee);
+                self.metrics.update_prioritization_fee(transaction_fee);
```
Comment (apfitzge) [inline runtime/src/prioritization_fee.rs:176]: Is this called for vote transactions as well? Those will all be unprioritized. Should those be separated?

File: runtime/src/prioritization_fee.rs [hunk 2]
```diff
@@ -29,6 +41,22 @@ impl PrioritizationFeeMetrics {
         saturating_add_assign!(self.total_update_elapsed_us, val);
     }
 
+    fn update_prioritization_fee(&mut self, fee: u64) {
+        if fee == 0 {
+            saturating_add_assign!(self.non_prioritized_transactions_count, 1);
```
Comment (apfitzge) [inline runtime/src/prioritization_fee.rs:46]: iirc from our earlier conversation, I think we wanted to get the range of non-zero fees. If that's the case, we should return early here.

**Comment by overclock-validator** [comment]
I kinda wonder what the best measure would be for the local feemarket-ness level/strength. I guess what would be indicative is if one of the more contended accounts has a high min priority relative to fees outside of that contended account. Maybe you could do some ratio of min priority of most highly accessed account vs base fee (or min priority fee outside of contended account). I think it's good to figure out some measurements to detect the level of local vs global fee market changes. You should be able to see a change in such a metric for example when decreasing account compute limits because it would increase local contention/congestion which should lead to a stronger local fee market.

**Comment by taozhu-chicago** [comment]
> https://discord.com/channels/428295358100013066/910937142182682656/1054466803532841011 Probably shouldn't be backported as it's not a critical fix?

Thought want to see that data from MB now. But good call, have removed the backport labels.

File: runtime/src/prioritization_fee.rs (hunk 1 above)
Comment (taozhu-chicago) [reply runtime/src/prioritization_fee.rs:176]: Ah, good point! Perhaps better to filter out votes from updating fee cache all together. in https://github.com/solana-labs/solana/pull/29341

File: runtime/src/prioritization_fee.rs (hunk 2 above)
Comment (taozhu-chicago) [reply runtime/src/prioritization_fee.rs:46]: OK, so we want to use non-fee/fee txs counters and fee range together. The fee range only applies to fee-paying txs. Updated comments and return early in df93cf58885b48b521e62c41bc1613a7673faed9

**Comment by taozhu-chicago** [comment]
> I kinda wonder what the best measure would be for the local feemarket-ness level/strength. I guess what would be indicative is if one of the more contended accounts has a high min priority relative to fees outside of that contended account. Maybe you could do some ratio of min priority of most highly accessed account vs base fee (or min priority fee outside of contended account). I think it's good to figure out some measurements to detect the level of local vs global fee market changes. You should be able to see a change in such a metric for example when decreasing account compute limits because it would increase local contention/congestion which should lead to a stronger local fee market. Having some standard of measurement could be helpful for adjusting compute limit parameters in the future.

Perhaps can do ratio = `max_prioritization_fee` / `min_prioritization_fee` ?

**Review by apfitzge — APPROVED** [review]
Happy with these added metrics; thanks Tao!

File: runtime/src/prioritization_fee.rs [hunk 3]
```diff
@@ -29,6 +41,24 @@ impl PrioritizationFeeMetrics {
...
+            return;
+        }
+
+        // update prioritized transaction fee metrics.
+        saturating_add_assign!(self.prioritized_transactions_count, 1);
+
+        if fee > self.max_prioritization_fee {
+            self.max_prioritization_fee = fee;
+        }
```
Comment (t-nelson) [inline runtime/src/prioritization_fee.rs:55]: might be a little cleaner as...

```suggestion
        self.max_prioritization_fee =
            self.max_prioritization_fee.max(fee);
```

File: runtime/src/prioritization_fee.rs [hunk 4]
```diff
@@ -29,6 +41,24 @@ impl PrioritizationFeeMetrics {
...
+
+        if self.min_prioritization_fee.is_none() || fee < self.min_prioritization_fee.unwrap() {
+            self.min_prioritization_fee = Some(fee);
+        }
+    }
```
Comment (t-nelson) [inline runtime/src/prioritization_fee.rs:60]: similarly, potentially cleaner as
```suggestion
        self.min_prioritization_fee = self
            .min_prioritization_fee
            .map_or(fee, |min_fee| min_fee.min(fee));
```

**Review by t-nelson — APPROVED** [review]
lgtm!  left some optional suggestions for cleaner code

File: runtime/src/prioritization_fee.rs (hunk 4 above)
Comment (apfitzge) [reply runtime/src/prioritization_fee.rs]: TIL about `map_or`, thanks!

**Pull request merged**